src/gstools/field/generator.py

```diff
@@ -532,3 +538,332 @@ def _create_unit_vector(self, broadcast_shape, axis=0):
+        shape : :class:`tuple`
+            the shape of the summed modes
+
+        Returns
+        -------
+        nugget : :class:`numpy.ndarray`
+            the nugget in the same shape as the summed modes
+        """
+        if self.model.nugget > 0:
+            nugget = np.sqrt(self.model.nugget) * self._rng.random.normal(
+                size=shape
+            )
+        else:
+            nugget = 0.0
+        return nugget
+
+    def update(self, model=None, seed=np.nan, mode_no=None, period=None):
+        """Update the model and the seed.
+
+        If model and seed are not different, nothing will be done.
+
+        Parameters
+        ----------
+        model : :any:`CovModel` or :any:`None`, optional
+            covariance model. Default: :any:`None`
+        seed : :class:`int` or :any:`None` or :any:`numpy.nan`, optional
+            the seed of the random number generator.
+            If :any:`None`, a random seed is used. If :any:`numpy.nan`,
+            the actual seed will be kept. Default: :any:`numpy.nan`
+        mode_no : :class:`list` or :any:`None`, optional
+            Number of Fourier modes per dimension.
+        period : :class:`list` or :any:`None, optional
+            The spatial periodicity of the field, is often the domain size.
+        """
+        tmp_model = model if model is not None else self._model
+        if period is not None:
+            if len(period) != tmp_model.dim:
+                raise ValueError(
+                    "Fourier: Dimension mismatch in argument period."
+                )
+            self._period = np.array(period)
+            self._delta_k = 2.0 * np.pi / self._period
+            if mode_no is None:
+                self._set_modes(self._mode_no, tmp_model)
+        if mode_no is not None:
+            if len(mode_no) != tmp_model.dim:
+                raise ValueError(
+                    "Fourier: Dimension mismatch in argument mode_no."
+                )
+            if (np.asarray([m % 2 for m in mode_no]) != 0).any():
+                raise ValueError("Fourier: Odd mode_no not supported.")
+            self._set_modes(mode_no, tmp_model)
+
+        # check if a new model is given
+        if isinstance(model, CovModel):
+            if self.model != model:
+                self._model = dcp(model)
+                if seed is None or not np.isnan(seed):
+                    self.reset_seed(seed)
+                else:
+                    self.reset_seed(self._seed)
+            # just update the seed, if its a new one
+            elif seed is None or not np.isnan(seed):
+                self.seed = seed
+        # or just update the seed, when no model is given
+        elif model is None and (seed is None or not np.isnan(seed)):
+            if isinstance(self._model, CovModel):
+                self.seed = seed
+            else:
+                raise ValueError(
+                    "gstools.field.generator.RandMeth: no 'model' given"
```

Comment:
Copypasta: Should be `Fourier` instead of `RandMeth`

Reply:
Thanks for pointing that out.